### ISSUE TITLE
Arakoon restarted without reason

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/arakoon#91]
Looking at the log file:

```
2016-06-30 14:20:52 968514 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235966 - info - establishing connection to (188.165.13.24,26413) (timeout = 5.000000)
2016-06-30 14:20:52 968765 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235967 - info - __open_connection to (188.165.13.24,26413) failed: Unix.Unix_error(Unix.ECONNREFUSED, "connect", ""); backtrace:; Raised at file "
src/unix/lwt_unix.ml", line 1419, characters 20-57; Called from file "src/unix/lwt_unix.ml", line 487, characters 13-24
2016-06-30 14:20:53 969780 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235968 - info - establishing connection to (188.165.13.24,26413) (timeout = 5.000000)
2016-06-30 14:20:53 969999 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235969 - info - __open_connection to (188.165.13.24,26413) failed: Unix.Unix_error(Unix.ECONNREFUSED, "connect", ""); backtrace:; Raised at file "src/unix/lwt_unix.ml", line 1419, characters 20-57; Called from file "src/unix/lwt_unix.ml", line 487, characters 13-24
2016-06-30 14:20:54 970793 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235970 - info - establishing connection to (188.165.13.24,26413) (timeout = 5.000000)
2016-06-30 14:20:54 971025 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235971 - info - __open_connection to (188.165.13.24,26413) failed: Unix.Unix_error(Unix.ECONNREFUSED, "connect", ""); backtrace:; Raised at file "src/unix/lwt_unix.ml", line 1419, characters 20-57; Called from file "src/unix/lwt_unix.ml", line 487, characters 13-24
2016-06-30 14:20:55 971856 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235972 - info - establishing connection to (188.165.13.24,26413) (timeout = 5.000000)
2016-06-30 14:20:55 972099 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235973 - info - __open_connection to (188.165.13.24,26413) failed: Unix.Unix_error(Unix.ECONNREFUSED, "connect", ""); backtrace:; Raised at file "src/unix/lwt_unix.ml", line 1419, characters 20-57; Called from file "src/unix/lwt_unix.ml", line 487, characters 13-24
2016-06-30 14:20:56 972218 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235974 - info - establishing connection to (188.165.13.24,26413) (timeout = 5.000000)
2016-06-30 14:20:56 972482 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235975 - info - __open_connection to (188.165.13.24,26413) failed: Unix.Unix_error(Unix.ECONNREFUSED, "connect", ""); backtrace:; Raised at file "src/unix/lwt_unix.ml", line 1419, characters 20-57; Called from file "src/unix/lwt_unix.ml", line 487, characters 13-24
2016-06-30 14:20:57 972949 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235976 - info - establishing connection to (188.165.13.24,26413) (timeout = 5.000000)
2016-06-30 14:20:57 973176 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235977 - info - __open_connection to (188.165.13.24,26413) failed: Unix.Unix_error(Unix.ECONNREFUSED, "connect", ""); backtrace:; Raised at file "src/unix/lwt_unix.ml", line 1419, characters 20-57; Called from file "src/unix/lwt_unix.ml", line 487, characters 13-24
2016-06-30 14:20:58 956135 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235978 - info - should roll over: tlog_entries=5000
2016-06-30 14:20:58 963411 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235979 - info - should roll: i= 45623944 old_outer = 110194 => jump to 110195
2016-06-30 14:20:58 963511 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235980 - info - closed 110194.tlog
2016-06-30 14:20:58 964457 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235981 - info - adding new task to compression queue /mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlog,/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlx
2016-06-30 14:20:58 964570 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235982 - info - Compressing: /mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlog into /mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlx
2016-06-30 14:20:58 964648 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235983 - info - Compressing "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlog" to "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlx" via "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlx.part"
2016-06-30 14:20:58 964672 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235984 - info - Unlinking "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs/110194.tlx.part"
2016-06-30 14:20:58 974143 +0200 - perf-roub-03 - 8538/0 - arakoon - 16235985 - info - establishing connection to (188.165.13.24,26413) (timeout = 5.000000)
2016-06-30 14:20:59 651237 +0200 - perf-roub-03 - 24678/0 - arakoon - 0 - info - --- NODE STARTED ---
2016-06-30 14:20:59 651267 +0200 - perf-roub-03 - 24678/0 - arakoon - 1 - info - git_revision: tags/1.9.6-0-gea32319 
2016-06-30 14:20:59 651276 +0200 - perf-roub-03 - 24678/0 - arakoon - 2 - info - compile_time: 10/06/2016 13:21:09 UTC 
2016-06-30 14:20:59 651283 +0200 - perf-roub-03 - 24678/0 - arakoon - 3 - info - version: 1.9.6
2016-06-30 14:20:59 651293 +0200 - perf-roub-03 - 24678/0 - arakoon - 4 - info - NOFILE: 1024
2016-06-30 14:20:59 651379 +0200 - perf-roub-03 - 24678/0 - arakoon - 5 - info - cluster_cfg={ cfgs = [{ node_name = "bKz9fJNRoh8ZGfe7"; ips = ["178.33.60.66"]; client_port = 26412; messaging_port = 26413; home = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/db"; tlog_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; log_sinks = [Console]; crash_log_sinks = [File(console:)]; tlx_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; head_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; log_level = "info"; log_config = None; batched_transaction_config = None; lease_period = 10; master = "Elected"; is_laggy = false; is_learner = false; is_witness = false; targets = []; compressor = Snappy; fsync = true; is_test = false; reporting = 300; _fsync_tlog_dir = true; node_tls = None; collapse_slowdown = None; head_copy_throttling = 0. }; { node_name = "nhfE8ZP77taJwMBG"; ips = ["178.33.62.163"]; client_port = 26412; messaging_port = 26413; home = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/db"; tlog_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; log_sinks = [Console]; crash_log_sinks = [File(console:)]; tlx_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; head_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; log_level = "info"; log_config = None; batched_transaction_config = None; lease_period = 10; master = "Elected"; is_laggy = false; is_learner = false; is_witness = false; targets = []; compressor = Snappy; fsync = true; is_test = false; reporting = 300; _fsync_tlog_dir = true; node_tls = None; collapse_slowdown = None; head_copy_throttling = 0. }; { node_name = "09FydVMSuGL4Fp3k"; ips = ["188.165.13.24"]; client_port = 26412; messaging_port = 26413; home = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/db"; tlog_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; log_sinks = [Console]; crash_log_sinks = [File(console:)]; tlx_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; head_dir = "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_1/tlogs"; log_level = "info"; log_config = None; batched_transaction_config = None; lease_period = 10; master = "Elected"; is_laggy = false; is_learner = false; is_witness = false; targets = []; compressor = Snappy; fsync = true; is_test = false; reporting = 300; _fsync_tlog_dir = true; node_tls = None; collapse_slowdown = None; head_copy_throttling = 0. }]; log_cfgs = []; batched_transaction_cfgs = []; _master = Elected; _lease_period = 10; cluster_id = "hdd_roub_nsm_1"; plugins = ["nsm_host_plugin"]; nursery_cfg = None; tlog_max_entries = 5000; tlog_max_size = 5000000; max_value_size = 8388608; max_buffer_size = 33554432; client_buffer_capacity = 32; lcnum = 16384; ncnum = 8192; tls = None }
2016-06-30 14:20:59 651391 +0200 - perf-roub-03 - 24678/0 - arakoon - 6 - info - Batched_store.max_entries = 100
2016-06-30 14:20:59 651397 +0200 - perf-roub-03 - 24678/0 - arakoon - 7 - info - Batched_store.max_size = 100000
2016-06-30 14:20:59 651405 +0200 - perf-roub-03 - 24678/0 - arakoon - 8 - info - autofix:true
2016-06-30 14:20:59 651412 +0200 - perf-roub-03 - 24678/0 - arakoon - 9 - info - loading plugin nsm_host_plugin
```

Looking at `syslog`:

```
Jun 30 10:12:50 perf-roub-03 kernel: [865141.531602] do_general_protection: 10 callbacks suppressed
Jun 30 10:12:50 perf-roub-03 kernel: [865141.531606] traps: arakoon[37790] general protection ip:ca4469 sp:7ffe7cc4e9d0 error:0 in arakoon[400000+a18000]
Jun 30 10:12:50 perf-roub-03 kernel: [865141.669490] init: ovs-arakoon-hdd_roub_nsm_0 main process (37790) killed by SEGV signal
Jun 30 10:12:50 perf-roub-03 kernel: [865141.669506] init: ovs-arakoon-hdd_roub_nsm_0 main process ended, respawning
```

When looking at `/var/crash` we see that an older logfile is kept then the given timestamp:

```
root@perf-roub-03:~# ls -alsh /var/crash/_usr_bin_arakoon.1000.crash
16M -rw-r----- 1 ovs ovs 16M Jun 24 09:17 /var/crash/_usr_bin_arakoon.1000.crash
```

**Comment by kinvaris** [comment]
Seeing this on the same node on another arakoon:

```
2016-06-30 10:12:48 922091 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852819 - info - Finished compression task, lets loop
2016-06-30 10:12:48 922100 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852820 - info - Taking job from compression queue...
2016-06-30 10:12:50 106991 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852821 - info - should roll over: tlog_entries=385
2016-06-30 10:12:50 111558 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852822 - info - should roll: i= 54649190 old_outer = 188054 => jump to 188055
2016-06-30 10:12:50 111701 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852823 - info - closed 188054.tlog
2016-06-30 10:12:50 115317 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852824 - info - adding new task to compression queue /mnt/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlo
gs/188054.tlog,/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlogs/188054.tlx
2016-06-30 10:12:50 115453 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852825 - info - Compressing: /mnt/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlogs/188054.tlog into /mnt
/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlogs/188054.tlx
2016-06-30 10:12:50 115535 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852826 - info - Compressing "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlogs/188054.tlog" to "/mnt
/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlogs/188054.tlx" via "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlogs/188054.tlx.part"
2016-06-30 10:12:50 115556 +0200 - perf-roub-03 - 37790/0 - arakoon - 1852827 - info - Unlinking "/mnt/hdd1/ovh/arakoon/hdd_roub_nsm_0/tlogs/188054.tlx.part"
2016-06-30 10:12:51 673086 +0200 - perf-roub-03 - 1241/0 - arakoon - 0 - info - --- NODE STARTED ---
2016-06-30 10:12:51 673131 +0200 - perf-roub-03 - 1241/0 - arakoon - 1 - info - git_revision: tags/1.9.6-0-gea32319 
2016-06-30 10:12:51 673145 +0200 - perf-roub-03 - 1241/0 - arakoon - 2 - info - compile_time: 10/06/2016 13:21:09 UTC 
2016-06-30 10:12:51 673158 +0200 - perf-roub-03 - 1241/0 - arakoon - 3 - info - version: 1.9.6
2016-06-30 10:12:51 673171 +0200 - perf-roub-03 - 1241/0 - arakoon - 4 - info - NOFILE: 1024
```

As seen in `syslog`:

```
Jun 30 10:12:50 perf-roub-03 kernel: [865141.531602] do_general_protection: 10 callbacks suppressed
Jun 30 10:12:50 perf-roub-03 kernel: [865141.531606] traps: arakoon[37790] general protection ip:ca4469 sp:7ffe7cc4e9d0 error:0 in arakoon[400000+a18000]
Jun 30 10:12:50 perf-roub-03 kernel: [865141.669490] init: ovs-arakoon-hdd_roub_nsm_0 main process (37790) killed by SEGV signal
Jun 30 10:12:50 perf-roub-03 kernel: [865141.669506] init: ovs-arakoon-hdd_roub_nsm_0 main process ended, respawning
```

**Comment by jeroenmaelbrancke** [comment]
Same issue on perf-roub-01:

```
Jul 19 04:17:24 perf-roub-01 kernel: [1303016.786146] traps: arakoon[73677] general protection ip:ca7a59 sp:7ffee5b52e70 error:0 in arakoon[400000+a1b000]
Jul 19 04:19:56 perf-roub-01 kernel: [1303169.093224] init: ovs-arakoon-hdd_roub_nsm_1 main process (73677) killed by SEGV signal
Jul 19 04:19:56 perf-roub-01 kernel: [1303169.093247] init: ovs-arakoon-hdd_roub_nsm_1 main process ended, respawning
```

Jira ticket: [ovs-3934](http://jira.openvstorage.com/browse/OVS-3934)

**Comment by domsj** [comment]
Compiler bug http://caml.inria.fr/mantis/view.php?id=7307 might very well be the cause of this...
This means we'll have to upgrade to ocaml 4.03.0 if we want to see this fixed.
Thanks for pointing me to the relevant mantis ticket @liegebeest!

**Comment by domsj** [comment]
should be resolved by #106
